WikiWatershed/mmw-geoprocessing
# Add soiln, soilp, recess_coeff operations

## Overview

Real-world MapShed requests also use these operations as part
of their modeling, so to better reflect the reality of usage
we add them to the example requests and responses.

Connects #83 

### Demo

Update benchmark:

```
> kj benchmark --mapshed

Timing HUC8 MultiOperation MapShed ->

HUC8 MultiOperation MapShed, run 1 -> 8.854033 s
HUC8 MultiOperation MapShed, run 2 -> 7.97886 s
HUC8 MultiOperation MapShed, run 3 -> 7.882735 s
HUC8 MultiOperation MapShed, run 4 -> 7.910098 s
HUC8 MultiOperation MapShed, run 5 -> 7.519246 s

HUC8 MultiOperation MapShed average -> 8.0289944 s

Timing HUC8 MapShed, separate requests per operation ->

HUC8 MapShed, 7 operations, run 1 -> 10.07 s
HUC8 MapShed, 7 operations, run 2 -> 10.46 s
HUC8 MapShed, 7 operations, run 3 -> 10.18 s
HUC8 MapShed, 7 operations, run 4 -> 9.74 s
HUC8 MapShed, 7 operations, run 5 -> 10.21 s
```

## Testing Instructions

* Checkout this branch and run `server`
* In another tab, run:

    ```shell
    $ http :8090/multi < examples/MultiOperationRequest.json | jq . > examples/MultiOperationResponse.json
    ```

  and then do `git status` to ensure nothing changed.
